jax/interpreters/partial_eval.py

```diff
@@ -699,24 +704,27 @@ def _remat_partial_eval(wrap_unknown_pvals, trace, _, f, tracers, params):
 
   # Using the instantiated tracers, run call_bind like JaxprTrace.process_call.
   in_pvals = [t.pval for t in instantiated_tracers]
-  with core.initial_style_staging():
-    jaxpr, eval_out_pvals, consts, env_tracers = trace.partial_eval(
-      f, in_pvals, partial(remat_call_p.bind, **params))
+  jaxpr, eval_out_pvals, consts, env_tracers = trace.partial_eval(
+    f, in_pvals, partial(remat_call_p.bind, **params))
+
+  # Convert consts to inputs, since they may contain Tracer instances.
+  jaxpr = convert_constvars_jaxpr(jaxpr)
+  const_tracers = map(trace.new_instantiated_const, consts)
```

Comment:
What's the benefit of doing this so early? It seems like it complicates the `partial_eval_jaxpr` part a little, because we also have to pass in consts as arguments.

Reply:
It's kind of specific to #3370, where the consts that come out might be tracers. Then it's easier just to treat them like tracers from the start, i.e. basically treat them like the env values.

Maybe we can iteratively simplify in follow-ups if we find ways to improve? My preference would be to iterate after landing #3370 though :)

Reply:
Sure, I don't mean to block this PR! It just seemed a little surprising.